packages/grid/x-data-grid/src/hooks/features/editRows/useGridCellEditing.new.ts

```diff
@@ -130,8 +130,18 @@ export const useGridCellEditing = (
         if (event.key === 'Escape') {
           reason = GridCellEditStopReasons.escapeKeyDown;
         } else if (event.key === 'Enter') {
+          // Wait until IME is settled for Asian languages like Japanese and Chinese
+          // TODO: `event.which` is depricated but this is a temporary workaround
+          if (event.which === 229) {
+            return;
+          }
           reason = GridCellEditStopReasons.enterKeyDown;
         } else if (event.key === 'Tab') {
+          // Wait until IME is settled for Asian languages like Japanese and Chinese
+          // TODO: `event.which` is depricated but this is a temporary workaround
+          if (event.which === 229) {
+            return;
+          }
```

Comment:
I think that we should wrap ALL the keys

```suggestion
```

Reply:
let me clarify. you mean 
```js
if (event.which === 229 && event.key === "Enter") { ... }
```
for enter button and so on? Is that correct?

Reply:
No, just move it up here so that it returns regardless the key

```diff
--- a/packages/grid/x-data-grid/src/hooks/features/editRows
/useGridCellEditing.new.ts
+++ b/packages/grid/x-data-grid/src/hooks/features/editRows
/useGridCellEditing.new.ts
@@ -125,6 +125,10 @@ export const useGridCellEditing = (
   const handleCellKeyDown = React.useCallback<GridEventLis
tener<'cellKeyDown'>>(
     (params, event) => {
       if (params.cellMode === GridCellModes.Edit) {
+        if (event.which === 229) {
+          return;
+        }
+
         let reason: GridCellEditStopReasons | undefined;
```

Same for row editing, and the old API (cell and row editing)

Reply:
oh make sense. Thank you!